WordPress/src/main/java/org/wordpress/android/ui/reader/repository/usecases/GetDiscoverCardsUseCase.kt

```diff
@@ -0,0 +1,48 @@
+package org.wordpress.android.ui.reader.repository.usecases
+
+import dagger.Reusable
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import org.wordpress.android.datasets.ReaderDiscoverCardsTable
+import org.wordpress.android.datasets.ReaderPostTable
+import org.wordpress.android.models.discover.ReaderDiscoverCard
+import org.wordpress.android.models.discover.ReaderDiscoverCard.InterestsYouMayLikeCard
+import org.wordpress.android.models.discover.ReaderDiscoverCard.ReaderPostCard
+import org.wordpress.android.models.discover.ReaderDiscoverCards
+import org.wordpress.android.modules.IO_THREAD
+import org.wordpress.android.ui.reader.ReaderConstants
+import javax.inject.Inject
+import javax.inject.Named
+
+@Reusable
+class GetDiscoverCardsUseCase @Inject constructor(
+    private val parseDiscoverCardsJsonUseCase: ParseDiscoverCardsJsonUseCase,
+    @Named(IO_THREAD) private val ioDispatcher: CoroutineDispatcher
+) : ReaderRepositoryDispatchingUseCase(ioDispatcher) {
```

Comment:
@malinajirka - I refactored out `ReaderRepositoryDispatchingUseCase` - may want to merge from develop